File: src/quest_editor/actions/EditShortDescriptionAction.ts
import { QuestEditAction } from "./QuestEditAction";

export class EditShortDescriptionAction extends QuestEditAction<string> {
    readonly description = "Edit short description";

    undo(): void {
        this.quest.set_short_description(this.old);
    }

    redo(): void {
        this.quest.set_short_description(this.new);
    }
}

File: web/src/main/kotlin/world/phantasmal/web/questEditor/actions/EditShortDescriptionAction.kt
package world.phantasmal.web.questEditor.actions

import world.phantasmal.web.core.actions.Action
import world.phantasmal.web.questEditor.models.QuestModel

class EditShortDescriptionAction(
    private val quest: QuestModel,
    private val newShortDescription: String,
    private val oldShortDescription: String,
) : Action {
    override val description: String = "Edit short description"

    override fun execute() {
        quest.setShortDescription(newShortDescription)
    }

    override fun undo() {
        quest.setShortDescription(oldShortDescription)
    }
}
